src/EntryPoints/PersistentPageIdFunction.php

```diff
@@ -28,8 +30,17 @@ public function handleParserFunctionCall( Parser $parser ): array {
 			return [];
 		}
 
+		$title = Title::castFromPageReference( $page );
+		$id = $this->getPersistentIdForPage( $page );
+
+		if ( $id === null && $title->exists() ) {
+			$pageId = $title->getArticleID();
+			$id = PersistentPageIdentifiersExtension::getInstance()->getIdGenerator()->generate();
+			$this->repo->savePersistentIds( [ $pageId => $id ] );
```

Comment:
The display function should not do write operations.

Reply:
Okay. So this is WIP and I am not sure yet what's considered a display hook/function. 

> Also, the side-effect of this code is that it creates persistent IDs for existing pages, which is not somethig we want. That's what the maintenance script is for.

We can easily prevent using  `!$wikiPage->isNew())`.


Purging the wiki page isn't working for me.  Was it for anyone else?

`onRevisionFromEditComplete` seems to get the job done. I'll fix tests after I know if it is okay to use `onRevisionFromEditComplete`.


https://github.com/user-attachments/assets/b48caa0c-b512-4c90-a5d0-efa532358ecf